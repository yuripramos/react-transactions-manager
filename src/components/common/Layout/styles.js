import styled, { css } from "styled-components";
import { media, rem } from "../../../styles/tools";

export const Content = styled.div`
  ${media.sm(css`
    min-height: calc(100vh - ${rem(150)});
  `)};
  ${media.md(css`
    min-height: calc(100vh - ${rem(180)});
  `)};
  ${media.lg(css`
    min-height: calc(100vh - ${rem(173)});
  `)};
`;
