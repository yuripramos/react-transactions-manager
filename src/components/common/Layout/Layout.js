import React, { Fragment } from "react";
import { node } from "prop-types";
import Header from "../../Header";

import { Content } from "./styles";

function Layout({ children }) {
  return (
    <Fragment>
      <Header />
      <Content>{children}</Content>
    </Fragment>
  );
}

Layout.propTypes = {
  children: node.isRequired // eslint-disable-line
};

export default Layout;
