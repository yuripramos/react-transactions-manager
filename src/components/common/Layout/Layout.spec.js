import React from "react";
import { shallow } from "enzyme";

import Layout from "./Layout";

describe("Layout component", () => {
  it("should match snapshot with children", () => {
    expect(
      shallow(
        <Layout>
          <h1>children</h1>
        </Layout>
      )
    ).toMatchSnapshot();
  });
});
