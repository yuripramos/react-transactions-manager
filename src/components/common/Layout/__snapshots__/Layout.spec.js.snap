// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot with children 1`] = `
<React.Fragment>
  <ConnectWrapper />
  <styled.div>
    <h1>
      children
    </h1>
  </styled.div>
</React.Fragment>
`;
